cmd/forwarder/config_test.go

```diff
@@ -0,0 +1,94 @@
+package main
+
+import (
+	"testing"
+)
+
+func TestIssConfig_LogAuthUser_NoValidUser(t *testing.T) {
+	var base = IssConfig{
+		TokenUserSamplePct: 100,
+	}
+
+	var hits int
+	for i := 0; i < 100000; i++ {
+		if base.LogAuthUser("user") {
+			hits++
+		}
+	}
+
+	if hits > 0 {
+		t.Errorf("expected no hits, but got %d\n", hits)
+	}
+}
```

Comment:
I have a strong aversion to tests that (a) repeat things, and (b) rely on randomness. I think there must be a way to refactor this so that we're not testing it this way. But I don't know what the right way to do it "in go" is.

One possibility is something like this:

```
type UserLogger struct {
  Ratio float 
  ValidTokenUser string
}

func (l UserLogger) log)(user u, float i) boolean {
  return l.ValidTokenUser != "" &&
	u != l.ValidTokenUser &&
	i <= l.Ratio
}
```

You could then just test this method in isolation, with something like `userLogger.log("user", 0.5)`, and make sure it returns true if the input is less than the ratio, and false otherwise.

Is this idiomatic `go`? I don't know.

Reply:
Actually that's even better.